Weekly_Work/Week2/Code_Changes_Git.py

```diff
@@ -0,0 +1,159 @@
+import json
+import datetime
+from Commit import Commit
+import pandas as pd
+import matplotlib.pyplot as plt
+import utils
+
+class Code_Changes_Git(Commit):
+    """Class for Code_Changes for Git repositories.
+    
+    Objects are instantiated by specifying a file with the
+    commits obtained by Perceval from a set of repositories.
+        
+    :param path: Path to file with one Perceval JSON document per line
+    """
+    
+    def __init__(self, path, date_range=(None, None), source_code_exclude_list=None):
+        """
+        Initilizes self.df, the dataframe with one commit per row.
+        The source_code_exclude_list parameter is a list which contains file extensions
+        which are to be ignored. All possible file extensions are allowed. 
+        For files without a standard ".xyz" extension, like LICENCE or AUTHORS, the "others" 
+        extension is used. 
+        
+        :param path: Path to file with one Perceval JSON document per line
+        :param date_range: A tuple which represents the start and end date of interest
+        :param source_code_exclude_list: Files extensions to exclude. For example, 
+            source_code_exclude_list = ["py", "other", "gitignore"]
+        """
+        
+        super().__init__(path, date_range, source_code_exclude_list)
+
+        
+    def total_count(self):
+        """
+        Get a naive count of the number of commits in the Perceval data.
+        Note that some commits may be repeated and so total_count may 
+        overshoot.
+        """
+        return len(self.df.index)
+    
+    def compute(self, empty=True, merge=True, master_only=False):
+        """Count number of commits of different types
+        
+        :param empty: Include empty commits
+        :param merge: Include merge commits
+        :param master_only: Include only commits made on master branch
+        """
+        count_per_repo = dict()
```

Comment:
As commented elsewhere, I don't think we need to count per repo... For that, we can just create several objects, one per repo, with the corresponding datasets. That won't be more ineeficient than this approach, and would be simpler.

Reply:
Fine. Will remove that.